File: mini-mvc/src/main/java/com/mini/core/security/digest/SHA512.java
package com.mini.core.security.digest;

import java.nio.charset.Charset;

public final class SHA512 extends BaseDigest {
	public SHA512() {
		super("SHA-512");
	}
	
	public static SHA512 getInstance() {
		return new SHA512();
	}
	
	public static String encode(String data, Charset charset) {
		return getInstance().update(data, charset).encode();
	}
	
	public static String encode(String data) {
		return getInstance().update(data).encode();
	}
}

File: mini-util/src/main/kotlin/com/mini/core/security/digest/SHA512.kt
@file:Suppress("unused")

package com.mini.core.security.digest

import java.nio.charset.Charset

class SHA512 : BaseDigest("SHA-512") {
   companion object {

       val instance: SHA512
           get() = SHA512()

       fun encode(data: String?, charset: Charset?): String {
           return instance.update(data!!, charset!!).encode()
       }

       fun encode(data: String?): String {
           return instance.update(data!!).encode()
       }
   }
}